### DISCUSSION TITLE
How to print a new line?

### DISCUSSION THREAD

**Discussion by ghost** [modula3/cm3#1138] [General]
On Pascal, there is WriteLn. What about CM3? I do not see PutLn.

**Comment by mikanystrom** [comment]
a new line?  "\n", like in C.

E.g., 

Wr.PutText(Stdio.stdout, "hello, world!\n")

Or 

IO.Put("hello, world!\n")

**Comment by RodneyBates** [comment]
On 1/10/23 22:14, Mika Nyström wrote:
>
> a new line? "\n", like in C.
>
> E.g.,
>
> Wr.PutText(Stdio.stdout, "hello, world!\n")
>
> Or
>
> IO.Put("hello, world!\n")
>

Or, instead of \n, you can use Wr.EOL, which is a
TEXT constant that is OS-specific to the target that
you compile for.  Thus, your same code, complied for
different targets, will produce the correct end of line
sequence for each target.  On some (Windows) this
is a two-character sequence, thus a TEXT and not a
CHAR.

e.g.:
   Wr.PutText(Stdio.stdout, "hello, world" & Wr.EOL)

or
    Wr.PutText(Stdio.stdout, "hello, world")
    Wr.PutText(Stdio.stdout, Wr.EOL)

As with most language libraries, don't forget, as I usually do,
to flush if you want to see the output immediately:

    Wr.Flush(Stdio.stdout)

RTIO.i3 has some alternative stderr-putting alternatives.

**Comment by mikanystrom-intel** [comment]
Ah I always wondered about Windows.  I am brainwashed by Unix.  Sorry, should have double-checked.

**Comment by VictorMiasnikov** [comment]
As smal remark:

IO.Put ("\n");

work good on OS Windows.

( It's tested with both cp866 and utf-8 code pages)

**Reply by ghost** [reply]
I confirm. I'm also using this syntax.
